Plugin/Block/Toolbar.php

```diff
@@ -0,0 +1,183 @@
+
+    /** @var AbstractCollection */
+    protected $_collection = null;
+
+    /**
+     * Toolbar constructor.
+     * @param Context $context
+     * @param NostoHelperData $nostoHelperData
+     * @param NostoHelperAccount $nostoHelperAccount
+     * @param StoreManagerInterface $storeManager
+     * @param CategoryBuilder $builder
+     * @param CategoryRecommendation $categoryRecommendation
+     * @param CookieManagerInterface $cookieManager
+     * @param NostoLogger $logger
+     * @param Registry $registry
+     * @param array $data
+     */
+    public function __construct(
+        Context $context,
+        NostoHelperData $nostoHelperData,
+        NostoHelperAccount $nostoHelperAccount,
+        StoreManagerInterface $storeManager,
+        CategoryBuilder $builder,
+        CategoryRecommendation $categoryRecommendation,
+        CookieManagerInterface $cookieManager,
+        NostoLogger $logger,
+        Registry $registry,
+        array $data = []
+    ) {
+        $this->nostoHelperData = $nostoHelperData;
+        $this->nostoHelperAccount = $nostoHelperAccount;
+        $this->categoryBuilder = $builder;
+        $this->storeManager = $storeManager;
+        $this->cookieManager = $cookieManager;
+        $this->categoryRecommendation = $categoryRecommendation;
+        $this->logger = $logger;
+        $this->registry = $registry;
+        parent::__construct($context, $data);
+    }
+
+    /**
+     * Plugin - Used to modify default Sort By filters
+     *
+     * @param MagentoToolbar $subject
+     * @param Collection $collection
+     * @return $this
+     * @throws NoSuchEntityException
+     */
+    public function afterSetCollection(
+        MagentoToolbar $subject,
+        $collection
+    ) {
+        $this->_collection = $collection;
+        $store = $this->storeManager->getStore();
+        $currentOrder = $subject->getCurrentOrder();
+        if (($currentOrder === Config::NOSTO_PERSONALIZED_KEY
+            || $currentOrder === Config::NOSTO_TOPLIST_KEY)
+            && $this->nostoHelperAccount->nostoInstalledAndEnabled($store)
+            && $this->nostoHelperData->isCategorySortingEnabled($store)
+        ) {
+            $limit = (int)$subject->getLimit();
+            if ($limit) {
+                $this->_collection->setPageSize($limit);
+            }
+            $currentPage = $subject->getCurrentPage();
+            $this->_collection->setCurPage($currentPage);
+
+            //Get ids of products to order
+            $orderIds = $this->getSortedIds($store, $currentOrder);
+            if (!empty($orderIds)) {
+                try {
+                    $zendExpression = new \Zend_Db_Expr('FIELD(e.entity_id,' . implode(',', $orderIds) . ') DESC');
+                    $this->_collection->getSelect()->order($zendExpression);
+                } catch (\Exception $e) {
+                    $this->logger->exception($e);
+                }
+            }
+
+            return $this;
```

Comment:
You're returning only inside the `if`. Is that intentional or you also want to return void? If so, void is missing in the docblock